src/Language/Haskell/Brittany/Internal/Layouters/Decl.hs

```diff
@@ -595,6 +612,92 @@ layoutPatternBindFinal alignmentToken binderDoc mPatDoc clauseDocs mWhereDocs ha
          ]
       ++ wherePartMultiLine
 
+
+--------------------------------------------------------------------------------
+-- TyClDecl
+--------------------------------------------------------------------------------
+
+layoutTyCl :: ToBriDoc TyClDecl
+layoutTyCl ltycl@(L _ tycl) = case tycl of
+#if MIN_VERSION_ghc(8,2,0)
+  SynDecl name vars fixity typ _ ->
+    docWrapNode ltycl $ layoutSynDecl name (hsq_explicit vars) fixity typ
+#else
+  SynDecl name vars typ _ ->
+    docWrapNode ltycl $ layoutSynDecl name (hsq_explicit vars) typ
+#endif
+  _ -> briDocByExactNoComment ltycl
+
+layoutSynDecl
+#if MIN_VERSION_ghc(8,4,0)
```

Comment:
I think `IdP GhcPs` ~ `GhcPs` ~ `RdrName`, so no #conditional necessary? Or am I messing this up?

Reply:
https://github.com/lspitzner/brittany/blob/89b02bc4e55cd7b5d384af0e464e365d1ae48d23/src/Language/Haskell/Brittany/Internal/Layouters/Decl.hs#L632-L636

Reply:
I thought that, but it would type check on `8.4.3` for some reason :confused:

Reply:
Ah, indeed. I think we can put

~~~~.hs
type family IdP p
type instance IdP GhcPs = RdrName
~~~~

to the <=8.2 conditional in `Prelude.hs` and avoid this instance of mid-signature conditional. I have tested this on all but 8.6.

(And I saw you cleaned up the conditionals a bit. Good work. Still not pretty, but I think that is the price for supporting multiple versions.)

Reply:
i.e. the type signature would be unconditionally `Located (IdP GhcPs)`

Reply:
Aha, I'll give that a go